actors/player/PlayerStateMachine.gd

```diff
@@ -37,7 +37,8 @@ func _unhandled_input(event):
 		_change_state('attack')
 		get_tree().set_input_as_handled()
 		return
-	current_state.handle_input(event)
+	if current_state:
```

Comment:
This shouldn't be part of the same commit as it's unrelated. Also, I have to check the code but I feel it's not the right fix to keep the code maintainable. There shouldn't be cases where the input gets forwarded to a nonexistent state.

Reply:
It's because the player performed `queue_free` so the old reference does not exist anymore.

Additionally _unhandled_input is an internal engine method so it can't be avoided without other changes. I would suggest not doing `queue_free` on the player and instead just setting to a Dead state.

Reply:
I have to get back into the code, the state machine should belong to the player and be freed with it. And yeah it probably shouldn't call queue_free upon dying then.